packages/merge/tests/merge-typedefs.spec.ts

```diff
@@ -1717,4 +1762,57 @@ describe('Merge TypeDefs', () => {
     const merged = mergeTypeDefs(typeDefs);
     expect(print(merged)).toBeSimilarString(print(directive));
   });
+
+  it('does not merge repeatable Federation directives without the same arguments', () => {
+    const ast = parse(/* GraphQL */ `
+      extend schema
+        @link(url: "https://specs.apollo.dev/link/v1.0")
+        @link(url: "https://specs.apollo.dev/federation/v2.6", import: ["@key"])
+
+      type Item @key(fields: "id") @key(fields: "id type") {
+        id: ID!
+        type: String!
+      }
+    `);
+    const merged = mergeTypeDefs([ast], { useSchemaDefinition: true });
+    expect(print(merged)).toBeSimilarString(print(ast));
+  });
+
+  it('understands imports if the import comes after the type definition', () => {
+    const ast = parse(/* GraphQL */ `
+      type Item @key(fields: "id") @key(fields: "id type") {
+        id: ID!
+        type: String!
+      }
+
+      extend schema
+        @link(url: "https://specs.apollo.dev/link/v1.0")
+        @link(url: "https://specs.apollo.dev/federation/v2.6", import: ["@key"])
+    `);
+    const merged = mergeTypeDefs([ast]);
+    expect(print(merged)).toBeSimilarString(print(ast));
+  });
+
+  it.todo('supports multiple schema extensions');
+  // , () => {
```

Comment:
this is outside the scope of this PR but i think it is something we need in order to better support federation, because `extend schema` with directives is common